google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/it/ITBigQueryWriteNonQuotaRetryTest.java

```diff
@@ -0,0 +1,277 @@
+    }
+  }
+
+  @Test
+  public void testJsonStreamWriterCommittedStreamWithNonQuotaRetry()
+      throws IOException, InterruptedException, ExecutionException,
+          DescriptorValidationException {
+    RetrySettings retrySettings =
+        RetrySettings.newBuilder()
+            .setInitialRetryDelay(Duration.ofMillis(500))
+            .setRetryDelayMultiplier(1.1)
+            .setMaxAttempts(5)
+            .setMaxRetryDelay(Duration.ofMinutes(1))
+            .build();
+    String tableName = "CommittedRetry";
+    TableId tableId = TableId.of(DATASET, tableName);
+    Field col1 = Field.newBuilder("col1", StandardSQLTypeName.STRING).build();
+    Schema schema = Schema.of(col1);
+    TableInfo tableInfo = TableInfo.newBuilder(tableId, StandardTableDefinition.of(schema)).build();
+    bigquery.create(tableInfo);
+    TableName parent = TableName.of(NON_QUOTA_RETRY_PROJECT_ID, DATASET, tableName);
+
+    WriteStream writeStream =
+        client.createWriteStream(
+            CreateWriteStreamRequest.newBuilder()
+                .setParent(parent.toString())
+                .setWriteStream(
+                    WriteStream.newBuilder().setType(WriteStream.Type.COMMITTED).build())
+                .build());
+    int totalRequest = 901;
+    int rowBatch = 1;
+    ArrayList<ApiFuture<AppendRowsResponse>> allResponses =
+        new ArrayList<ApiFuture<AppendRowsResponse>>(totalRequest);
+    try (JsonStreamWriter jsonStreamWriter =
+        JsonStreamWriter.newBuilder(writeStream.getName(), writeStream.getTableSchema())
+            .setRetrySettings(retrySettings)
+            .build()) {
+      for (int k = 0; k < totalRequest; k++) {
+        JSONObject row = new JSONObject();
+        row.put("col1", "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa");
+        JSONArray jsonArr = new JSONArray();
+        // 3MB batch.
+        for (int j = 0; j < rowBatch; j++) {
+          jsonArr.put(row);
+        }
+        LOG.info("Appending: " + k + "/" + totalRequest);
+        allResponses.add(jsonStreamWriter.append(jsonArr, k * rowBatch));
+      }
+      LOG.info("Waiting for all responses to come back");
+      for (int i = 0; i < totalRequest; i++) {
+        LOG.info("Waiting for request " + i);
+        try {
+          Assert.assertEquals(
+              allResponses.get(i).get().getAppendResult().getOffset().getValue(), i * rowBatch);
+        } catch (ExecutionException ex) {
+          Assert.fail("Unexpected error " + ex);
+        }
+      }
+    }
+  }
+
+  @Test
+  public void testJsonStreamWriterDefaultStreamWithNonQuotaRetry()
+      throws IOException, InterruptedException, ExecutionException,
+          DescriptorValidationException {
+    RetrySettings retrySettings =
+        RetrySettings.newBuilder()
+            .setInitialRetryDelay(Duration.ofMillis(500))
+            .setRetryDelayMultiplier(1.1)
+            .setMaxAttempts(5)
+            .setMaxRetryDelay(Duration.ofMinutes(1))
```

Comment:
Is this Delay useful for nonquotaretry? And the Multiplier useful?

Reply:
It's not useful for non-quota retry.  Same for multiplier.  Removed